app/jobs/connect_app_job.rb

```diff
@@ -20,7 +20,7 @@ def perform(app, token, channel: nil)
     selector = "#connected-app-discover-regen.enterprise_#{enterprise.id}"
     html = ApplicationController.render(
       partial: "admin/enterprises/form/connected_apps/discover_regen",
-      locals: { enterprise:, connected_app: enterprise.connected_apps.first},
+      locals: { enterprise:, connected_app: enterprise.connected_apps.discover_regen.first },
```

Comment:
I guess that the more generic alternative would be `ConnectedApp.of(enterprise).first`. We'll see with more usage what will be better.